### PR TITLE
[WIP] Processor plugin

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

This adds a processor plugin and a builtin registry.

We should move the plugin interface into the SDK to simplify things. Then the plugin doesn't have to import Conduit and Conduit doesn't have to translate objects when sending them to the plugin (at least if it's a builtin plugin). This makes the architecture quite a bit faster and simpler than what we currently have with connector plugins.

Related to #1285

### Quick checks:

- [X] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [X] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [X] I have written unit tests.
- [X] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/plugin/processor/plugin.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	// Parameters.
+	Description string
+	// Version string. Should be a semver prepended with `v`, e.g. `v1.54.3`.
+	Version string
+	// Author declares the entity that created or maintains this plugin.
+	Author string
+	// SourceParams and DestinationParams are maps of named Parameters that
+	// describe how to configure the plugins Destination or Source.
+	SourceParams      map[string]Parameter
+	DestinationParams map[string]Parameter
```
Comment (hariso) [inline pkg/plugin/processor/plugin.go:80]: This comes probably from the connector SDK, but I believe we need only params for processors.

File: pkg/plugin/processor/plugin.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	"context"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+)
+
+// ProcessorPlugin receives records, manipulates them and returns back the processed
+// records.
+//
+//nolint:revive // This interface will be moved into the SDK
+type ProcessorPlugin interface {
```
Comment (hariso) [inline pkg/plugin/processor/plugin.go:29]: As per the conversation we had earlier: https://github.com/ConduitIO/conduit-processor-sdk/pull/1

Comment (hariso) [reply pkg/plugin/processor/plugin.go:29]: The PR above was merged.

**Comment by lovromazgon** [comment]
Closing in favor of #1322

**Pull request closed without merging**